apps/src/templates/certificates/petition/PetitionForm.jsx

```diff
@@ -0,0 +1,101 @@
+import React, {useState} from 'react';
+import {Button} from 'react-bootstrap';
+import {range, mapValues} from 'lodash';
+import {
+  buildControlledFieldGroup,
+  createErrorMessage,
+  keyValidation
+} from '@cdo/apps/templates/certificates/petition/PetitionHelpers';
+
+const PetitionForm = () => {
+  // data starts with all required fields having an empty value to ensure proper validation
+  const [data, setData] = useState(mapValues(keyValidation, () => ''));
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const handleChange = event => {
+    event.persist();
+    setData(data => ({...data, [event.target.name]: event.target.value}));
+  };
+
+  const handleSubmit = e => {
+    e.preventDefault();
+    setErrorMessage(createErrorMessage(data));
+  };
+
+  const buildFieldGroup = (
+    id,
+    placeholderOrLabel,
+    helpText,
+    componentClass,
+    children
+  ) =>
+    buildControlledFieldGroup(
+      id,
+      placeholderOrLabel,
+      helpText,
+      componentClass,
+      children,
+      handleChange,
+      data
+    );
+
+  return (
+    <>
+      <form
+        id="petition-form"
+        className="petition-form"
+        onSubmit={handleSubmit}
+      >
+        <div className={'petition-space'}>{errorMessage}</div>
+        {buildFieldGroup('name', 'Name')}
+        {buildFieldGroup('email', 'Email', 'Only used for infrequent updates')}
+        {buildFieldGroup(
+          'zip-or-country',
+          'ZIP code or country',
+          'Enter country if outside the United States'
+        )}
+        {buildFieldGroup(
+          'age',
+          'Age',
+          <a href="/privacy">See our privacy practices for children</a>,
+          'select',
+          ['-', ...range(1, 101)].map((age, index) => (
+            <option key={index} value={age}>
+              {age}
+            </option>
+          ))
+        )}
+        {buildFieldGroup(
+          'profession',
+          'I am a',
+          <a href="/privacy">See our privacy practices for children</a>,
```

Comment:
it looks like we're including this text twice (looks a little weird on the page)

Reply:
Ah jeez. Got a little too copy and paste happy! Good catch